restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/mybatis/MybatisExtensionsConfigurationDao.java

```diff
@@ -276,4 +283,86 @@ private ExtensionConfiguration toExtensionConfiguration(final Map<String, Object
         map.put("enabled", extensionConfiguration.isEnabled());
         return map;
     }
+
+    @Override
+    public ExtensionConfiguration getAccountExtensionConfiguration(String accountSid, String extensionSid) {
+        final SqlSession session = sessions.openSession();
+        ExtensionConfiguration extensionConfiguration = null;
+        try {
+            Map<String, Object> params = new HashMap<String, Object>();
+            params.put("account_sid", accountSid.toString());
+            params.put("extension_sid", extensionSid.toString());
+            final Map<String, Object> result  = session.selectOne(namespace + "getAccountExtensionConfiguration", params);
+            if (result != null) {
+                extensionConfiguration = toAccountsExtensionConfiguration(result);
+            }
+            return extensionConfiguration;
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public void addAccountExtensionConfiguration(ExtensionConfiguration extensionConfiguration, Sid accountSid) throws ConfigurationException {
+        final SqlSession session = sessions.openSession();
+        try {
+            if (extensionConfiguration != null && extensionConfiguration.getConfigurationData() != null) {
+                if (validate(extensionConfiguration)) {
+                    final Map<String, Object> map = new HashMap<String, Object>();
+                    map.put("account_sid", DaoUtils.writeSid(accountSid));
+                    map.put("extension_sid", DaoUtils.writeSid(extensionConfiguration.getSid()));
+
+                    if (extensionConfiguration.getConfigurationData() != null)
```

Comment:
@gvagenas Please have another look.
-I have addressed most of the comments
-I have added the common Extension handler functions to ExtensionController but have not done this for CallManager.
-I have moved also `DefaultExtensionConfiguration` from telscale to community
-I have deferred the REST design + implementation issues to be addressed in #2127. I think we should discuss about the design at first.
-I have deferred from adding the extension hook for "MESSAGE" use case as this seems to be a new scenario which was not even encountered during testing. Will open an Issue for this